python/xorbits/_mars/dataframe/base/pivot.py

```diff
@@ -0,0 +1,300 @@
+
+        .. versionchanged:: 1.1.0
+            Also accept list of index names.
+
+    values : str, object or a list of the previous, optional
+        Column(s) to use for populating new frame's values. If not
+        specified, all remaining columns will be used and the result will
+        have hierarchically indexed columns.
+
+    Returns
+    -------
+    DataFrame
+        Returns reshaped DataFrame.
+
+    Raises
+    ------
+    ValueError:
+        When there are any `index`, `columns` combinations with multiple
+        values. `DataFrame.pivot_table` when you need to aggregate.
+
+    See Also
+    --------
+    DataFrame.pivot_table : Generalization of pivot that can handle
+        duplicate values for one index/column pair.
+    DataFrame.unstack : Pivot based on the index values instead of a
+        column.
+    wide_to_long : Wide panel to long format. Less flexible but more
+        user-friendly than melt.
+
+    Notes
+    -----
+    For finer-tuned control, see hierarchical indexing documentation along
+    with the related stack/unstack methods.
+
+    Reference :ref:`the user guide <reshaping.pivot>` for more examples.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({'foo': ['one', 'one', 'one', 'two', 'two',
+    ...                            'two'],
+    ...                    'bar': ['A', 'B', 'C', 'A', 'B', 'C'],
+    ...                    'baz': [1, 2, 3, 4, 5, 6],
+    ...                    'zoo': ['x', 'y', 'z', 'q', 'w', 't']})
+    >>> df
+        foo   bar  baz  zoo
+    0   one   A    1    x
+    1   one   B    2    y
+    2   one   C    3    z
+    3   two   A    4    q
+    4   two   B    5    w
+    5   two   C    6    t
+
+    >>> df.pivot(index='foo', columns='bar', values='baz')
+    bar  A   B   C
+    foo
+    one  1   2   3
+    two  4   5   6
+
+    >>> df.pivot(index='foo', columns='bar')['baz']
+    bar  A   B   C
+    foo
+    one  1   2   3
+    two  4   5   6
+
+    >>> df.pivot(index='foo', columns='bar', values=['baz', 'zoo'])
+            baz       zoo
+    bar   A  B  C   A  B  C
+    foo
+    one   1  2  3   x  y  z
+    two   4  5  6   q  w  t
+
+    You could also assign a list of column names or a list of index names.
+
+    >>> df = pd.DataFrame({
+    ...        "lev1": [1, 1, 1, 2, 2, 2],
+    ...        "lev2": [1, 1, 2, 1, 1, 2],
+    ...        "lev3": [1, 2, 1, 2, 1, 2],
+    ...        "lev4": [1, 2, 3, 4, 5, 6],
+    ...        "values": [0, 1, 2, 3, 4, 5]})
+    >>> df
+        lev1 lev2 lev3 lev4 values
+    0   1    1    1    1    0
+    1   1    1    2    2    1
+    2   1    2    1    3    2
+    3   2    1    2    4    3
+    4   2    1    1    5    4
+    5   2    2    2    6    5
+
+    >>> df.pivot(index="lev1", columns=["lev2", "lev3"], values="values")
+    lev2    1         2
+    lev3    1    2    1    2
+    lev1
+    1     0.0  1.0  2.0  NaN
+    2     4.0  3.0  NaN  5.0
+
+    >>> df.pivot(index=["lev1", "lev2"], columns=["lev3"], values="values")
```

Comment:
Something wrong with this test case.

```python
In [15]: df = pd.DataFrame({
    ...:        "lev1": [1, 1, 1, 2, 2, 2],
    ...:        "lev2": [1, 1, 2, 1, 1, 2],
    ...:        "lev3": [1, 2, 1, 2, 1, 2],
    ...:        "lev4": [1, 2, 3, 4, 5, 6],
    ...:        "values": [0, 1, 2, 3, 4, 5]}, chunk_size=3
    ...:        )

In [16]: df.pivot(index="lev1", columns=["lev2", "lev3"], values="values")
Out[16]:
lev2  1       2
lev3  2  1    2    1
lev1
2     3  4  5.0  NaN
1     1  0  NaN  2.0

In [17]: df.to_pandas().pivot(index="lev1", columns=["lev2", "lev3"], values="values")
Out[17]:
lev2    1         2
lev3    1    2    1    2
lev1
1     0.0  1.0  2.0  NaN
2     4.0  3.0  NaN  5.0
```

Reply:
Understood. The index is more twisted than `groupby`, making it confusing at the first glance.